### PR TITLE
Remove UGX from zero decimal currencies

### LINKED ISSUE #6946: UGX generates zero decimal currency errors because should not be zero based
### Describe the bug

UGX is now a zero decimal currency, and placing orders in UGX generates errors. 

### To Reproduce

1. Go to shop base settings and set UGX as base currency
2. Set up a product using UGX and checkout
3. Checkout

### Actual behavior

With UPE disabled a general error occurs `We cannot process at this time`

### Expected behavior

UGX is supported, as it shows as available as a base currency

### Additional context
6626828-zd-a8c

### PR DESCRIPTION
**Pull request by naman03malhotra**
Fixes #6946 

#### Changes proposed in this Pull Request
Removed UGX currency from ZDC `zero_decimal_currencies` array.

#### Testing instructions
- Setup a store with base currency UGX. Ensure that the products are setup with correct UGX prices (e.g. 18 UGX is not a valid price. The UGX to USD conversion rate is 4000)
- Use multi-currency module to checkout with UGX/ISK.
- Test checkout for different types of currencies - zero-decimal, special case (UGX, ISK) and two decimal currencies.
- Test Transaction, Order Edit and other screens if the values shown are correct.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/class-wc-payments-utils.php [hunk 1]
```diff
@@ -202,7 +202,6 @@ public static function zero_decimal_currencies(): array {
 			'mga', // Malagasy Ariary.
 			'pyg', // Paraguayan Guaraní.
 			'rwf', // Rwandan Franc.
-			'ugx', // Ugandan Shilling.
```
Comment (naman03malhotra) [inline includes/class-wc-payments-utils.php:205]: Not sure if any changes needs to be made here https://github.com/Automattic/woocommerce-payments/blob/e6e7c3be3d06153233bd14159bb2c4f0f1e120b4/i18n/locale-info.php#L3667

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7575` or branch name `fix/6946-remove-ugx-zero-decimal` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6946-remove-ugx-zero-decimal) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 53ceb0842e52056afe563964e3d6e58a08888230
- Build time: 2023-11-10 14:11:16 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by zmaglica** [comment]
I have tested it and I didn't get the error from issue

**Review by zmaglica — APPROVED** [review]

File: includes/class-wc-payments-utils.php (hunk 1 above)
Comment (zmaglica) [reply includes/class-wc-payments-utils.php:205]: I think not since it's a Stripe custom rule for this specific currency.

**Comment by naman03malhotra** [comment]
Closing in favor of https://github.com/Automattic/woocommerce-payments/pull/7620

**Review by tpaksu — APPROVED** [review]
I've tested only buying with UGX, and in some (unimportant) places there's `.00` appended to the price, like the order confirmation page. 

Cart, checkout, blocks checkout show the correct prices without decimals attached.

Order received page:
image

My account orders list:
image

My Account order details:
image

Orders list:
image

Order Details:
image

Transaction list:
image

Transaction details:
image

LGTM!

**Review by jessy-p — APPROVED** [review]
Added changelog and tested the basic flows in UGX, looks good.

**Comment by Brianmitchtay** [comment]
I see this PR was merged to [i18n-20231112](https://github.com/Automattic/woocommerce-payments/releases/tag/i18n-20231112), is the plan for this to be released to a production version soon so that users can take advantage of the fix? 

We have a merchant that is inquiring after this issue, and debating using WooPayments, right now a fix is released in a production version of the Stripe gateway but appears not to be in WooPayments. I'd love to be able to help them get set up on WooPayments, but that won't be likely until a fix for this is released in a production version.

Edit, merchant in 7220457-zen

**Comment by tpaksu** [comment]
Hello @Brianmitchtay, this will be released within the scheduled next version (6.8.0) in 3 days (Nov 16).

**Pull request merged**